packages/components/src/Combobox/hooks/useComboboxValidation.ts

```diff
@@ -1,58 +1,70 @@
-} from "../components/ComboboxTrigger";
-import { ComboboxContent } from "../components/ComboboxContent";
+  ComboboxOption,
+  ComboboxOptionProps,
+} from "../components/ComboboxOption/ComboboxOption";
+import { ComboboxAction } from "../components/ComboboxAction";
+import {
+  ComboboxActionProps,
+  ComboboxActivatorProps,
+  ComboboxProps,
+} from "../Combobox.types";
 import { ComboboxActivator } from "../components/ComboboxActivator";
 
 export const COMBOBOX_TRIGGER_COUNT_ERROR_MESSAGE =
-  "Combobox can only have one Trigger or Activator element";
-export const COMBOBOX_REQUIRED_CHILDREN_ERROR_MESSAGE =
-  "Combobox must have a Combobox.Content element";
+  "Combobox must have exactly one Trigger element";
 
-export function useComboboxValidation(children: ReactNode): {
-  triggerElement: ReactNode;
-  contentElement: ReactNode;
+export function useComboboxValidation(children?: ComboboxProps["children"]): {
+  triggerElement?: ReactElement;
+  optionElements?: ReactElement[];
+  actionElements?: ReactElement[];
 } {
-  const childrenArray = React.Children.toArray(children);
-  let triggerElement: ReactNode,
-    contentElement: ReactNode,
-    multipleTriggersFound = false;
-
-  childrenArray.forEach(child => {
-    if (isTriggerElement(child)) {
-      if (triggerElement) {
-        multipleTriggersFound = true;
-      }
-      triggerElement = child;
-    }
-
-    if (isContentElement(child)) {
-      contentElement = child;
-    }
-  });
+  const optionElements = getCompoundComponents<ComboboxOptionProps>(
+    ComboboxOption,
+    children,
+  );
+  const actionElements = getCompoundComponents<ComboboxActionProps>(
+    ComboboxAction,
+    children,
+  );
+  const activatorElements = getCompoundComponents<ComboboxActivatorProps>(
+    ComboboxActivator,
+    children,
+  );
 
-  useAssert(multipleTriggersFound, COMBOBOX_TRIGGER_COUNT_ERROR_MESSAGE);
+  const shouldThrowTriggerError = isInvalid(activatorElements);
 
-  useAssert(!contentElement, COMBOBOX_REQUIRED_CHILDREN_ERROR_MESSAGE);
+  useAssert(shouldThrowTriggerError, COMBOBOX_TRIGGER_COUNT_ERROR_MESSAGE);
 
   return {
-    contentElement,
-    triggerElement,
+    optionElements,
+    triggerElement: activatorElements[0],
+    actionElements,
   };
 }
 
-function isTriggerElement(child: ReactNode): boolean {
-  return (
-    React.isValidElement(child) &&
-    (child.type === ComboboxTriggerButton ||
-      child.type === ComboboxTriggerChip ||
-      child.type === ComboboxActivator)
-  );
+function isInvalid(
+  activators: ReactElement<
+    ComboboxActivatorProps,
+    string | React.JSXElementConstructor<ComboboxActivatorProps>
+  >[],
+): boolean {
+  return activators.length > 1 ? true : false;
 }
 
-function isContentElement(child: ReactNode): boolean {
-  return React.isValidElement(child) && child.type === ComboboxContent;
+/**
+ * Return all instances child component that matches the `type` provided
+ */
+export function getCompoundComponents<T>(
```

Comment:
originally I was borrowing this from DataListUtils

then we had to update the types to allow for mapping with the whole `ComboboxNode` meaning the `getCompoundComponents` now needed to accept `ComboboxNode` as a type for `children` which is waay to tied to Combobox

with that, I've copy/pasted a variation of it here. I don't like this very much, but I haven't had the time to think about how to make this generic & uncoupled from `ComboxboxNode` which is the type of `children` we must deal with